swagger.yaml

```diff
@@ -25,365 +327,653 @@ paths:
-                  name: E-Coin
-                  group_slug: crypto-currency
-                  delivery_team: Dark Warriors
+          description: List of Products
+          schema:
+            $ref: '#/definitions/ProductList'
+        401:
+          description: UNAUTHORIZED
 
     post:
       tags: [Product]
-      operationId: app.handler.product.add
+      operationId: app.handler.products.create
+      description: Create a new Product
       parameters:
-        - $ref: '#/parameters/ProductDef'
+        - name: product
+          in: body
+          required: true
+          schema:
+            type: object
+            properties:
+              name:
+                type: string
+                description: The name for the new Product
+                example: ZMON
+              product_group_uri:
+                type: string
+                description: Product Group URI
+                example: /product-groups/110
+            required:
+              - name
+              - product_group_uri
       responses:
         201:
-          description: Product created
+          description: CREATED
+          schema:
+            $ref: '#/definitions/Product'
+        400:
+          description: BAD REQUEST
+        401:
+          description: UNAUTHORIZED
+        404:
+          description: NOT FOUND (Product group does not exist)
+
+  /products/{id}:
+    get:
+      tags: [Product]
+      summary: Retrieve a Product
+      operationId: app.handler.products.get
+      parameters:
+        - $ref: '#/parameters/ResourceId'
+      responses:
+        200:
+          description: OK.
+          schema:
+            $ref: '#/definitions/Product'
+        401:
+          description: UNAUTHORIZED.
+        404:
+          description: NOT FOUND.
+
+    put:
+      tags: [Product]
+      operationId: app.handler.products.update
+      summary: Update a Product
+      parameters:
+        - $ref: '#/parameters/ResourceId'
+        - name: product
+          in: body
+          required: true
+          schema:
+            $ref: '#/definitions/Product'
+      responses:
+        200:
+          description: OK
+          schema:
+            $ref: '#/definitions/Product'
         400:
           description: BAD REQUEST
+        401:
+          description: UNAUTHORIZED
         404:
-          description: Product group Not Found
+          description: NOT FOUND
 
-  /products/{product}:
     delete:
       tags: [Product]
-      operationId: app.handler.product.delete
+      summary: Delete a Product
+      operationId: app.handler.products.delete
       parameters:
-        - $ref: '#/parameters/Product'
+        - $ref: '#/parameters/ResourceId'
+      responses:
+        204:
+          description: NO CONTENT.
+        401:
+          description: UNAUTHORIZED.
+        404:
+          description: NOT FOUND.
+
+  /products/{product_id}/sli:
+    get:
+      tags: [SLI]
+      description: Get list of Product SLIs
```

Comment:
The main idea is that SLI can be referenced in multiple SLO ``Target``. In this proposal the SLI def will include the ``source`` as well, which doesn't change. If we coupled SLI (along with their sources) to SLO targets, then we will lose reusability and we end up repeating the same SLI def in different SLO targets.